// @slack/web-api/dist/types/response/ConversationsOpenResponse.js
"use strict";
/* eslint-disable */
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// !!! DO NOT EDIT THIS FILE !!!                                                       //
//                                                                                     //
// This file is auto-generated by scripts/generate-web-api-types.sh in the repository. //
// Please refer to the script code to learn how to update the source data.             //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElementType = exports.DescriptionType = void 0;
var DescriptionType;
(function (DescriptionType) {
    DescriptionType["Mrkdwn"] = "mrkdwn";
    DescriptionType["PlainText"] = "plain_text";
})(DescriptionType || (exports.DescriptionType = DescriptionType = {}));
var ElementType;
(function (ElementType) {
    ElementType["Broadcast"] = "broadcast";
    ElementType["Channel"] = "channel";
    ElementType["Color"] = "color";
    ElementType["Date"] = "date";
    ElementType["Emoji"] = "emoji";
    ElementType["Link"] = "link";
    ElementType["Team"] = "team";
    ElementType["Text"] = "text";
    ElementType["User"] = "user";
    ElementType["Usergroup"] = "usergroup";
})(ElementType || (exports.ElementType = ElementType = {}));
